New boarding mission (#4735)
endless-sky/endless-sky: data/boarding missions.txt

	destination "Earth"

mission "Asteroid Course"
	assisting
	minor
	name "Passengers to <planet>"
	description "Transport the remaining crew of a destroyed ship, <bunks> people that were working on the <origin>, to <destination>."
	to offer
		random < 6
	source
		government "Merchant" "Syndicate"
		category "Light Freighter" "Interceptor"
		not attributes "automaton"
	passengers 3 7
	destination 
		distance 1

> 		not government "Pirate"

Reviewer: Should the destination include `attributes shipyard`?

Author: added

	on offer
		conversation
			`As you are approaching the <origin>, your radar sensors detect that the ship you are assisting is about to be slammed by a huge asteroid! While in a normal flight this wouldn't cause much trouble, a ship that has already suffered heavy damages and cannot dodge could be completely destroyed.`
			branch unarmed
				not "armament deterrence"
			choice
				`	(Shoot the asteroid.)`
					goto fire
				`	(Maneuver my ship to shield the <origin>.)`
					goto move
				`	(Attempt to repair the <origin> so that it can dodge the asteroid.)`
					goto repair
				`	(Interrupt the docking maneuvers and fly away.)`
					goto escape
			label unarmed
			choice
				`	(Maneuver my ship to shield the <origin>.)`
					goto move

> 				`	(Try to quickly repair the <origin>, so that it can move away from the asteroid's course.)`

Reviewer: This choice doesn't match the same on in the armed choices.

					goto repair
				`	(Interrupt the docking maneuvers and fly away.)`
			label escape

> 			`	You turn your ship around and activate your thrusters, trying to get as far as you can from the disabled ship. A few seconds later, the asteroid tears the <origin> apart, killing all of its crew members.`

Reviewer: ```suggestion
			`	You turn your ship around and activate your thrusters, trying to get as far as you can from the disabled ship. A few seconds later, the asteroid collides with the <origin>, tearing it to shreds and killing all of its crew members while leaving little more than a scratch on the asteroid's surface.`
```

				depart
			label fire
			branch success
				random > 50

> 			`	The shots you fire quickly break the asteroid into pieces, which keep following the same course. Before you can even try to destroy them, most have already hit the <origin>, which is now quickly depressurizing and losing the hull's integrity. There's not much you can do anymore, except to try to move a safe distance from the exploding ship...`

Reviewer: You committed my suggestion then deleted it in the mission compression. :(
```suggestion
			`	You fire at the asteroid in an attempt to destroy it, but it looks like you've only made things worse. The asteroid has now broken up into dozens of smaller chunks still flying directly for the <origin>. Before you can even react to fix your mistake, most of the chunks have already hit the <origin>, which is now quickly depressurizing. There's nothing more you can do except move a safe distance away from the exploding ship...`
```

				depart
			label success

> 			`	The shots you fire quickly break the asteroid into pieces, which seem to have changed course enough to completely miss the <origin>. After repairing it, you hail its captain, which thanks you with a payment of 30,000 credits.`

Reviewer: ```suggestion
			`	The shots you fire quickly break the asteroid into pieces, which seem to have changed course enough to completely miss the <origin>. After repairing it, you hail its captain, who thanks you with a payment of 30,000 credits.`
```

				decline
			label repair
			branch crew
				random > 60
			`	You dock with the <origin>, hoping to get out of this alive, but as soon as you reach the hatch, the asteroid hits and a series of explosions start coming from the disabled ship. You are able to reach your cockpit and undock, but the crew on the other ship couldn't reach yours in time...`
				depart
			label crew

> 			`	You dock with the <origin>, hoping to get out of this alive: but, as soon as you reach the hatch, the asteroid hits and a series of explosions start coming from the disabled ship. Most of the other ship's crew has been able to reach your ship by the time you undock from it, and they ask you to take them to <destination>.`

Reviewer: ```suggestion
			`	You dock with the <origin>, hoping to get out of this alive, but, as soon as you reach the hatch, the asteroid hits and a series of explosions start coming from the disabled ship. Most of the other ship's crew has been able to reach your ship by the time you undock from it, and they ask you to take them to <destination>.`
```

			choice
				`	"Okay, it's not far."`

> 					accept

Reviewer: ```suggestion
					launch
```

				`	"How much can you pay me?"`
			`	"I have a few hundred thousand credits," one of them says, presumably the captain. "I need to save enough to buy a new starship, but you will still receive a decent pay. And it's just a jump away from here anyways."`
				launch
			label move

> 			`	The asteroid, after hitting your shield, breaks into a million fragments which fly away from the <origin>. After repairing it, you hail its captain, which thanks you with a payment of 30,000 credits.`

Reviewer: ```suggestion
			`	The asteroid, after hitting your shield, breaks into a million fragments which fly away from the <origin>. After repairing it, you hail its captain, who thanks you with a payment of 30,000 credits.`
```

mission "Asteroid Course A"
	assisting
	minor
	to offer
		random < 10
	source
		government "Merchant" "Syndicate"
		category "Light Freighter" "Interceptor"
		not attributes "automaton"
	destination "Earth"
	on offer
		conversation

> 			`As you are approaching the <origin>, your radar sensors detect that the ship you are assisting is in the middle of a large asteroid's course! While in a normal flight this wouldn't cause much trouble, a ship without active shields and that already suffered heavy damages could be completely destroyed.`

Reviewer 2: Change "that already" to "that has already"

Author: Done

> 			`As you are approaching the <origin>, your radar sensors detect that the ship you are assisting is in the middle of a large asteroid's course! While in a normal flight this wouldn't cause much trouble, a ship without active shields and that has already suffered heavy damages could be completely destroyed.`

Reviewer: I feel like this could be changed up a bit to increase the sense of urgency. Instead of saying that the ship is in the middle of the asteroid's course, say that the ship is about the be impacted/slammed by an asteroid, or that you spot a large asteroid coming right for the ship. Then for the explanation, I wouldn't say that it's because the ship is without shields and has already taken damage, because asteroids never impact the ship normally. Instead say something like since the ship is disabled it can't dodge.

			branch unarmed

> 				"armament deterrence" == 0

Reviewer: ```suggestion
				not "armament deterrence"
```

> 				has "armament deterrence" == 0

Reviewer 3: This condition should get flagged as unrecognized / invalid. Either `not "armament deterrence"` or `armament deterrence == 0` is what you want.

Author: Weird, the 'has "armament deterrence" == 0' worked fine on my end and gave no error. I know it applies to all the fleet, but that was the best I could think of and it works decently for what I need here.
Removed the "has" anyways

			choice

> 				`	(Shoot towards the asteroid to destroy it.)`

Reviewer: "towards" is superfluous here. I'd drop "to destroy it" because we can't guarantee that are guns will actually do the job.
```suggestion
				`	(Shoot the asteroid.)`
```

					goto fire

> 				`	(Move your ship between the asteroid and the <origin>, then use your shields to protect it.)`

Reviewer: Action choices are first person, not second, so this should read "Move my ship..." Maybe also change this to something more along the lines of `(Maneuver my ship to shield the <origin>.)`, as that shortens this up while saying the same thing.

					goto move

> 				`	(Try to quickly repair the <origin>, so that it can move away from the asteroid's course.)`

Reviewer: ```suggestion
				`	(Attempt to repair the <origin> so that it can dodge the asteroid.)`
```

					goto repair
				`	(Interrupt the docking maneuvers and fly away.)`
					goto escape
			label unarmed
			choice
				`	(Move your ship between the asteroid and the <origin>, then use your shields to protect it.)`
					goto move
				`	(Try to quickly repair the <origin>, so that it can move away from the asteroid's course.)`
					goto repair
				`	(Interrupt the docking maneuvers and fly away.)`
			label escape
			`	You turn your ship around and activate your thrusters, trying to get as far as you can from the disabled ship. A few seconds later, the asteroid tears the <origin> apart, killing all of its crew members.`

> 				launch

Reviewer: This should be `flee` or `depart`. Otherwise we're accepting this mission to go to Earth. Same story for the two below this. Since you're using `on decline`, you'd want to use `depart`, as that defers and doesn't trigger the `on decline`.

			label fire

> 			`	The shots you fire quickly break the asteroid into pieces, which keep following the same course: before you can even try to destroy them, most have already hit the <origin>, which is now quickly depressurizing and losing the hull's integrity. There's not much you can do anymore, except to try moving to safe distance from it...`

Reviewer 2: Split the first sentence into two; with the colon, it gets a bit too long for my taste.
Change "try moving to safe distance from it" to "try to move a safe distance from the exploding ship"

Author: done

> 			`	The shots you fire quickly break the asteroid into pieces, which keep following the same course. Before you can even try to destroy them, most have already hit the <origin>, which is now quickly depressurizing and losing the hull's integrity. There's not much you can do anymore, except to try to move a safe distance from the exploding ship...`

Reviewer: ```suggestion
			`	You fire at the asteroid in an attempt to destroy it, but it looks like you've only made things worse. The asteroid has now broken up into dozens of smaller chunks still flying directly for the <origin>. Before you can even react to fix your mistake, most of the chunks have already hit the <origin>, which is now quickly depressurizing. There's nothing more you can do except move a safe distance away from the exploding ship...`
```

				launch
			label repair

> 			`	You dock with the <origin>, hoping to get out of this alive: but, as soon as you reach the hatch, the asteroid hits and a series of explosions start coming from the disabled ship. You are able to reach the cockpit and undock, but the crew on the other ship couldn't reach yours in time...`

Reviewer 2: Change "alive: but, as soon" to "alive, but as soon"
Maybe clean up the second sentence a bit? When I first read it, I thought that the player went to the disabled ship's cockpit, until I got to the bit about the other ship not making it.

Author: Changed both

			branch unarmed
				has "armament deterrence" == 0
			choice
				`	(Shoot towards the asteroid to destroy it.)`
					goto fire
				`	(Move your ship between the asteroid and the <origin>, then use your shields to protect it.)`
					goto move
				`	(Try to quickly repair the <origin>, so that it can move away from the asteroid's course.)`
					goto repair
				`	(Interrupt the docking maneuvers and fly away.)`
					goto escape
			label unarmed
			choice
				`	(Move your ship between the asteroid and the <origin>, then use your shields to protect it.)`
					goto move
				`	(Try to quickly repair the <origin>, so that it can move away from the asteroid's course.)`
					goto repair
				`	(Interrupt the docking maneuvers and fly away.)`
			label escape
			`	You turn your ship around and activate your thrusters, trying to get as far as you can from the disabled ship. A few seconds later, the asteroid tears the <origin> apart, killing all of its crew members.`
				launch
			label fire

> 			`	The shots you fire quickly break the asteroid into pieces, which seem to have changed course enough to completely miss the <origin>. After repairing it, you hail its captain, which thanks you with a payment of 100,000 credits`

Reviewer 4: I don't see a corresponding payment line.

Author: fixed, I forgot to set it to decline

> 			`	The shots you fire quickly break the asteroid into pieces, which seem to have changed course enough to completely miss the <origin>. After repairing it, you hail its captain, which thanks you with a payment of 30,000 credits`

Reviewer 5: missing period

> 				decline

Reviewer: Instead of having an entirely separate mission that only changes this single outcome, you can put the following under `label fire` in the first mission.
```
branch success
	random > 50
```

					goto move
				`	(Try to quickly repair the <origin>, so that it can move away from the asteroid's course.)`
					goto repair
				`	(Interrupt the docking maneuvers and fly away.)`
					goto escape
			label unarmed
			choice
				`	(Move your ship between the asteroid and the <origin>, then use your shields to protect it.)`
					goto move
				`	(Try to quickly repair the <origin>, so that it can move away from the asteroid's course.)`
					goto repair
				`	(Interrupt the docking maneuvers and fly away.)`
			label escape
			`	You turn your ship around and activate your thrusters, trying to get as far as you can from the disabled ship. A few seconds later, the asteroid tears the <origin> apart, killing all of its crew members.`
				launch
			label fire
			`	The shots you fire quickly break the asteroid into pieces, which keep following the same course. Before you can even try to destroy them, most have already hit the <origin>, which is now quickly depressurizing and losing the hull's integrity. There's not much you can do anymore, except to try to move a safe distance from the exploding ship...`
				launch

> 			label repair

Reviewer: Again, you can use a branch in the original mission that rolls a random number instead of needing an entirely separate mission.

			`	You dock with the <origin>, hoping to get out of this alive: but, as soon as you reach the hatch, the asteroid hits and a series of explosions start coming from the disabled ship. Most of the other ship's crew has been able to reach your ship by the time you undock from it, and they ask you to take them to <destination>.`
			choice
				`	"Okay, it's not far."`
					accept
				`	"How much can you pay me?"`

> 			`	"I have a few hundred thousand credits," one of them says, presumibly the captain. "I need to save enough to buy a new starship, but you will still recive a decent pay. And it's just a jump away from here anyways."`

Reviewer 2: presumably instead of presumibly

Author: Done

> 			`	"I have a few hundred thousand credits," one of them says, presumably the captain. "I need to save enough to buy a new starship, but you will still recive a decent pay. And it's just a jump away from here anyways."`

Reviewer 3: `recive`
